tempesta_fw/str.h

```diff
@@ -209,20 +209,24 @@ size_t tfw_ultoa(unsigned long ai, char *buf, unsigned int len);
  * @flags	- 3 most significant bytes for number of chunks of compound
  * 		  string and the least significant byte for flags;
  */
-typedef struct {
-	void		*ptr;
+typedef struct tfwstr_t {
+	union {
+		char *data;
+		struct tfwstr_t *chunks;
+		size_t num;
```

Comment:
Er, the field `num` is introduced to eliminate extra type conversions in recent changes by @i-rinat . But storing unsigned integer in `TfwStr->ptr` is a hack, the string is not supposed to be used until the pointer is set to some valid value. The hack is reasonable there and it's a kind of well-known hack. But I don't want to legalise it on architecture level: allowing arbitrary integer numbers can't be distinguished from the pointers, so any time we want to use the pointer, we need to check some flag in the `TfwStr` to be sure that the pointer is valid.  It's to huge penalty to support one single well known hack.

Reply:
`num` field was removed.